### PR TITLE
#167313419 Users should be able to like a specific comment

### PR DESCRIPTION
**Pull request by minega25**
#### What does this PR do?
- Users should be able to like a specific comment
#### Description of Task to be completed?
Users should be able to like a specific comment
#### How should this be manually tested?
Endpoints to be tested:
- `GET api/v1/comments/{commentId}`
- `POST api/v1/comments/{commentId}`
- `PUT api/v1/comments/{commentId}`
Run below commands to get started:
- `git checkout -b ft-user-like-comment-167313419`
- `git pull origin ft-user-like-comment-167313419`
- `npm run dev`
#### What are the relevant pivotal tracker stories?
[#167313419](https://www.pivotaltracker.com/story/show/167313419)
#### Screenshots (if appropriate)
Screen Shot 2019-08-29 at 11 46 57
Screen Shot 2019-08-29 at 11 39 00
Screen Shot 2019-08-29 at 11 39 44
Screen Shot 2019-08-29 at 11 44 55
Screen Shot 2019-08-29 at 11 45 34

### CODE REVIEW

**Review by salviosage — CHANGES_REQUESTED** [review]
please rebase.

**Review by nkalyesubula — DISMISSED** [review]
LGTM

**Comment by minega25** [comment]
> please rebase.

Worked on this @salviosage

**Review by habinezadalvan — DISMISSED** [review]
Great job @minega25, I went through your work and I found it convincing with nice algorithm.

**Review by nkalyesubula — DISMISSED** [review]

**Review by salviosage — DISMISSED** [review]
LGTM

**Review by nshutijonathan — DISMISSED** [review]
LGTM

File: src/controllers/comments.controller.js [hunk 1]
```diff
@@ -140,5 +142,104 @@ class Comments {
...
+    if (!comment) {
+      util.setError(404, `Comment with id: ${id} does not exist.`);
+      return util.send(res);
+    }
+    const { likesCount, likeInfo } = comment;
+
+    const userHasLikedBefore = likeInfo.search(username);
+
+    if (userHasLikedBefore === -1) {
+      util.setError(200, { data: { likesCount, likeInfo } });
```
Comment (Mnickii) [inline src/controllers/comments.controller.js:237]: Better would be ```if (!userHasLikedBefore)```

**Review by Mnickii — APPROVED** [review]
LGTM, but a comment

File: src/controllers/comments.controller.js (hunk 1 above)
Comment (minega25) [reply src/controllers/comments.controller.js:237]: @Mnickii this search function returns -1 when no match was found. Which interns makes it a `truthy` value when being evaluated. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search#Return_value

When we find a match, a positive number is returned as well being `truthy` too.

What can we do in this instance?

Comment (Mnickii) [reply src/controllers/comments.controller.js:237]: oh cool, i see likeInfo is a list of the users who've liked it. In this case then it's alright. @minega25

Comment (minega25) [reply src/controllers/comments.controller.js:237]: I have tried using the not operator the condition is not being met

**Pull request merged**